bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/callmonitor/CallMonitor.java

```diff
@@ -106,7 +113,7 @@ public void run() {
                     reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
                     // reset the retry interval
                     reconnectTime = 60000L;
-                } catch (Exception e) {
+                } catch (IOException e) {
```

Comment:
Since this is a custom thread uncaught runtime exceptions will not be logged if they occur. To make troubleshooting easier I think that you should add an UncaughtExceptionHandler to the thread so that unexpected thread deaths can be logged.

Reply:
Something like https://github.com/openhab/openhab-addons/pull/9734/commits/153347848b8de4941ae421d4b074d71de86dc8c3?